sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java

```diff
@@ -522,10 +543,14 @@ public String getName() {
 
   private TestProcessContext<InputT, OutputT> createProcessContext(
       OldDoFn<InputT, OutputT> fn,
-      InputT elem) {
+      TimestampedValue<InputT> elem) {
+    WindowedValue<InputT> windowedValue = elem.getTimestamp() == null
+      ? WindowedValue.valueInGlobalWindow(elem.getValue())
+      : WindowedValue.timestampedValueInGlobalWindow(elem.getValue(), elem.getTimestamp());
```

Comment:
If you're replacing the timestamp with a non-null value you should instead just use the last line here.